commit 3c3d818a9317a5f8ff7fb4547b2d6d5bcc39137d
Author: Cristian Marussi <cristian.marussi@arm.com>
Date:   Wed Jan 18 12:14:21 2023 +0000

    firmware: arm_scmi: Add core raw transmission support
    
    Add SCMI raw mode support which exposes a userspace interface to allow for
    bare SCMI command injection and snooping from userspace.
    
    Signed-off-by: Cristian Marussi <cristian.marussi@arm.com>
    Tested-by: Florian Fainelli <f.fainelli@gmail.com>
    Tested-by: Vincent Guittot <vincent.guittot@linaro.org>
    Link: https://lore.kernel.org/r/20230118121426.492864-13-cristian.marussi@arm.com
    Signed-off-by: Sudeep Holla <sudeep.holla@arm.com>

diff --git a/drivers/firmware/arm_scmi/driver.c b/drivers/firmware/arm_scmi/driver.c
index 5d5defd92403..2b96d9c3489d 100644
--- a/drivers/firmware/arm_scmi/driver.c
+++ b/drivers/firmware/arm_scmi/driver.c
@@ -920,7 +920,8 @@ static void scmi_handle_response(struct scmi_chan_info *cinfo,
 	trace_scmi_msg_dump(info->id, cinfo->id, xfer->hdr.protocol_id,
 			    xfer->hdr.id,
 			    xfer->hdr.type == MSG_TYPE_DELAYED_RESP ?
-			    "DLYD" : "RESP",
+			    (!SCMI_XFER_IS_RAW(xfer) ? "DLYD" : "dlyd") :
+			    (!SCMI_XFER_IS_RAW(xfer) ? "RESP" : "resp"),
 			    xfer->hdr.seq, xfer->hdr.status,
 			    xfer->rx.buf, xfer->rx.len);
 
@@ -1045,7 +1046,8 @@ static int scmi_wait_for_reply(struct device *dev, const struct scmi_desc *desc,
 			/* Trace polled replies. */
 			trace_scmi_msg_dump(info->id, cinfo->id,
 					    xfer->hdr.protocol_id, xfer->hdr.id,
-					    "RESP",
+					    !SCMI_XFER_IS_RAW(xfer) ?
+					    "RESP" : "resp",
 					    xfer->hdr.seq, xfer->hdr.status,
 					    xfer->rx.buf, xfer->rx.len);
 		}